include/psimd.h

```diff
@@ -530,6 +530,81 @@
 		return (psimd_f32) (mask & (psimd_s32) v);
 	}
 
+	/* Vector xor */
+	PSIMD_INTRINSIC psimd_s8 psimd_xor_s8(psimd_s8 a, psimd_s8 b) {
+		return a ^ b;
+	}
+
+	PSIMD_INTRINSIC psimd_u8 psimd_xor_u8(psimd_u8 a, psimd_u8 b) {
+		return a ^ b;
+	}
+
+	PSIMD_INTRINSIC psimd_s16 psimd_xor_s16(psimd_s16 a, psimd_s16 b) {
+		return a ^ b;
+	}
+
+	PSIMD_INTRINSIC psimd_u16 psimd_xor_u16(psimd_u16 a, psimd_u16 b) {
+		return a ^ b;
+	}
+
+	PSIMD_INTRINSIC psimd_s32 psimd_xor_s32(psimd_s32 a, psimd_s32 b) {
+		return a ^ b;
+	}
+
+	PSIMD_INTRINSIC psimd_u32 psimd_xor_u32(psimd_u32 a, psimd_u32 b) {
+		return a ^ b;
+	}
+
+	/* Vector shift right */
+	PSIMD_INTRINSIC psimd_s8 psimd_shr_s8(psimd_s8 a, uint32_t b) {
+		return a >> b;
```

Comment:
Clang before 4.0 requires both parts of a `>>` operator to be vectors, so you should use `a >> psimd_splat_s8((int8_t) b)`